pkg/operator/ceph/cluster/mon/spec.go

```diff
@@ -317,6 +317,15 @@ func (c *Cluster) makeMonDaemonContainer(monConfig *monConfig) corev1.Container
 		WorkingDir:    config.VarLogCephDir,
 	}
 
+	if monConfig.Zone != "" {
+		desiredLocation := fmt.Sprintf("%s=%s", c.stretchFailureDomainName(), monConfig.Zone)
+		container.Args = append(container.Args, []string{"--set-crush-location", desiredLocation}...)
+		if monConfig.Zone == c.getArbiterZone() {
+			// remember the arbiter mon to be set later in the reconcile after the OSDs are configured
+			c.arbiterMon = monConfig.DaemonName
+		}
+	}
+
```

Comment:
Do we need to do a check and enforce that the user is using v16.2.5 or higher here? That might be good.

Reply:
Actually I'll add a note in the pending release notes that stretch requires 16.2.5 or higher. I can't add a code check because it will block downstream from running on a custom build of nautilus for stretch clusters.

Reply:
We could do the check but have an env var option that would disable the check for downstream. I think providing clear feedback for upstream users who try to use the wrong version is a good practice.

Reply:
Stretch clusters are a much less common scenario where I haven't seen upstream feedback, so I think we'll be fine upstream without a check.

Reply:
I think it's acceptable to have different checks for upstream and downstream. I agree stretch mode is downstream focus but I don't want downstream to drive upstream code especially if it brings clarity to users. Also, since it's now considered stable We do this already for rbd and cephfs mirror which caused some small issues downstream, we just need to track this so during the resync we change the version for downstream Nautilus.

Reply:
Agreed downstream shouldn't drive the upstream since upstream is first. I'll go ahead and add the check, then we can remove it downstream as needed.

Reply:
Added as a new commit